tests/isoObject.cpp

```diff
@@ -0,0 +1,407 @@
+{
+	// Get system time
+	struct timespec currentTime;
+	clock_gettime(CLOCK_REALTIME, &currentTime);
+	time->tv_sec = currentTime.tv_sec;
+	time->tv_usec = currentTime.tv_nsec / 1000;
+	return time;
+}
+
+class SimulatedTestObject : public TestObject {
+	public:
+	SimulatedTestObject(int tcpSocket) : TestObject(tcpSocket) {
+		this->startHandleTCP();
+		state->handleEvent(*this, ISO22133::Events::B);
+	}
+	int handleUDPMessage(char *buffer, int bufferLen, int udpSocket, char *addr, const uint32_t port) {
+		if (isAwaitingFirstHeab()) {
+			std::stringstream ss;
+			ss.str(std::string());
+			ss << "Received UDP data from ATOS" << std::endl;
+			std::cout << ss.str();
+			this->setProcessChannelEndpoint(udpSocket, addr, port);
+			startSendMonr(); // UDP connection established, start sending MONR
+		}
+		return handleMessage(buffer, bufferLen);
+	}
+};
+
+class ATOSEmulator
+{
+	public:
+		ATOSEmulator(const std::string& ip = "0.0.0.0", int id = 1, int transmitterID = -1, char messCnt = 0) : 
+		listenIP(ip),
+		tcpSocket(context),
+		udpSocket(context, ip::udp::v4()),
+		receiverID(id),
+		messageCounter(messCnt),
+		transmitterID(transmitterID) {
+		}
+		~ATOSEmulator() {
+
+		}
+		void connect() {
+			try {
+				tcpSocket.connect(ip::tcp::endpoint(ip::address_v4::from_string(listenIP), ISO_22133_DEFAULT_OBJECT_TCP_PORT));
+			} catch (boost::system::system_error err) {
+				EXPECT_EQ(1, 2) << "TCP Connection failed";
+			}
+		}
+		void sendTCP(const std::vector<char>& data) {
+			tcpSocket.send(buffer(data));
+		}
+		void sendUDP(const std::vector<char>& data) {
+			ip::udp::endpoint receiverEP(ip::address_v4::from_string(listenIP), ISO_22133_OBJECT_UDP_PORT);
+			udpSocket.send_to(buffer(data), receiverEP);
+		}
+		void disconnect() {
+			try {
+				tcpSocket.close();
+			} catch (boost::system::system_error err) {
+				std::cout << "Error closing TCP socket" << std::endl;
+			}
+			try {
+				udpSocket.close();
+			} catch (boost::system::system_error err) {
+				std::cout << "Error closing UDP socket" << std::endl;
+			}
+		}
+		std::size_t receiveUDP(std::vector<char>& data, ip::udp::endpoint& ep) {
+			return udpSocket.receive_from(buffer(data), ep);
+		}
+
+		void receiveTCP(std::vector<char>& data) {
+			tcpSocket.receive(buffer(data));
+		}
+
+		void sendHeartbeat(const ControlCenterStatusType ccStatus) {
+			HeabMessageDataType heartbeat;
+			TimeSetToCurrentSystemTime(&heartbeat.dataTimestamp);
+			heartbeat.controlCenterStatus = ccStatus;
+			std::vector<char> transmitBuffer(1024);
+			MessageHeaderType header;
+			header.receiverID = this->receiverID;
+			header.messageCounter = this->messageCounter;
+			header.transmitterID = this->transmitterID;
+			auto nBytesWritten = encodeHEABMessage(&header, &heartbeat.dataTimestamp, heartbeat.controlCenterStatus,
+								transmitBuffer.data(), transmitBuffer.size(), false);
+			transmitBuffer.resize(nBytesWritten);
+			sendUDP(transmitBuffer);
+		}
+
+		void sendOSEM() {
```

Comment:
Would like to see this split in to two or renamed to buildAndSendOSEM or similar. Preferably split to follow single responsibility.